caravel/assets/javascripts/SqlLab/components/SouthPane.jsx

```diff
@@ -1,25 +1,42 @@
 import { Alert, Tab, Tabs } from 'react-bootstrap';
 import QueryHistory from './QueryHistory';
 import ResultSet from './ResultSet';
-import { areArraysShallowEqual } from '../../reduxUtils';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import * as Actions from '../actions';
 import React from 'react';
 
 import shortid from 'shortid';
 
+// editorQueries are queries executed by users
+// passed from SqlEditor component
+// queries are all queries executed
+// passed from global state for the purpose of data preview tabs
 const propTypes = {
-  queries: React.PropTypes.array.isRequired,
+  editorQueries: React.PropTypes.array.isRequired,
```

Comment:
Originally we pass in queries that are executed in SqlEditors here. With data preview in South Pane we have to listen to all queries in state, so I changed name of queries passed from SqlEditor as 'editorQueries', while use 'queries' from all queries mapped from global state.